docs/NOSAI_MASTER_ROADMAP_CANONICAL_V1.md

```
@@ -0,0 +1,439 @@
+| `nosai/runtime/nostale_observer.py` | read-only live observation | EYES input and World Model feed |
+| `nosai/runtime/telemetry.py` | bounded sequence/session telemetry | Perception Frame sequencing and device sync |
+| `nosai/runtime/adapter.py` + sandbox | provider-neutral runtime boundary | future runtime execution gate |
+| `app/dashboard/*` | dashboard server, events, observability, runtime bridge and state | PC dashboard + shared control concepts for mobile UI |
+| `app/autoset.py` + `app/hardware_profile.py` | hardware detection and deterministic benchmark | PC runtime tuning and future mobile capability profile |
+| `.github/workflows/*` | CI, quality, security, Test Center, benchmark, Windows/runtime workflows | release gates for every new module |
+
+**Rule:** do not create a second memory system, second telemetry system, second decision contract, second evaluation report format, or second safety gate when an existing component already covers the need.
+
+## 4. A-to-Z functional roadmap
+
+### A — Architecture & Contracts
```

markdownlint-cli2 (0.23.2)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)